### DISCUSSION TITLE
Edge function / bandwidth usage exploded past two days with no change in traffic

### DISCUSSION THREAD

**Discussion by aplchian** [vercel/community#342] [Help] [answered]
Our edge / _middleware function usage has exploded the past 2 days with no real change in traffic. Is there any way yall can think to troubleshoot the cause of this? please help :(

![image](https://user-images.githubusercontent.com/7388748/153464870-7d3e31d4-ce81-4460-be38-cb1c403743a4.png)
![image](https://user-images.githubusercontent.com/7388748/153465045-2521422d-6153-40ae-a8f2-cb28f37c7731.png)

**Comment by Pieparker** [comment, marked as answer]
Always a little tricky to debug project usage without any way to identify the project! Do you have the project ID to hand? Though, these spikes are the result in an increase in traffic.

**Reply by aplchian** [reply]
Hi!

prj_pBFNUUC7QK5rZkCMs4xG5RWNeTIU

**Reply by Pieparker** [reply]
What you'll want to do is configure [log drains](https://vercel.com/support/articles/how-do-i-store-logs-on-vercel) to keep tabs on usage and understand where the usage is derived from.

Armed with this information, you will understand whether requests are erroneous or malicious. From there, you can handle them however you wish! Examples may include:

- [Redirecting based on a User-Agent or another identifiable header such as IP](https://github.com/vercel/community/discussions/74#discussioncomment-2118409), if someone or something is spamming your site/functions
- Amending your Edge Function to redirect/not perform its action, similarly to above
- Working with our sales team to understand our Enterprise offering (check out the FAQ [here](https://vercel.com/security))

I hope this helps!
